src/pybind/mgr/mgr_module.py

```diff
@@ -920,6 +940,184 @@ def cluster_log(self, channel: str, priority: ClusterLogPrio, message: str) -> N
     def version(self) -> str:
         return self._version
 
+    def pool_exists(self, name: str) -> bool:
+        pools = [p['pool_name'] for p in self.get('osd_map')['pools']]
+        return name in pools
+
+    def have_enough_osds(self) -> bool:
+        # wait until we have enough OSDs to allow the pool to be healthy
+        ready = 0
+        for osd in self.get("osd_map")["osds"]:
+            if osd["up"] and osd["in"]:
+                ready += 1
+
+        need = cast(int, self.get_ceph_option("osd_pool_default_size"))
+        return ready >= need
+
+    def rename_pool(self, name: str, newname: str) -> None:
+        result = CommandResult('')
+        self.send_command(result, 'mon', '', json.dumps({
+            'prefix': 'osd pool rename',
+            'format': 'json',
+            'srcpool': name,
+            'destpool': newname,
+        }), '')
+        r, outb, outs = result.wait()
+        if r is not 0:
+            raise RuntimeError("cannot create \"{pool}\" pool")
+
+    def create_pool(self, name: str) -> None:
+        result = CommandResult('')
+        self.send_command(result, 'mon', '', json.dumps({
+            'prefix': 'osd pool create',
+            'format': 'json',
+            'pool': name,
+            'pg_num': 1,
+            'pg_num_min': 1,
+        }), '')
+        r, outb, outs = result.wait()
+        if r is not 0:
+            raise RuntimeError("cannot create \"{pool}\" pool")
+
+    def appify_pool(self, name: str, app: str) -> None:
+        result = CommandResult('')
+        self.send_command(result, 'mon', '', json.dumps({
+            'prefix': 'osd pool application enable',
+            'format': 'json',
+            'pool': name,
+            'app': app,
+        }), '')
+        r, outb, outs = result.wait()
+        if r is not 0:
+            raise RuntimeError(f"cannot set \"{name}\" pool application")
```

Comment:
Instead  of `send_command` use `check_mon_command`. It checks the return value

Reply:
Thanks for pointing that out -- fixed!